### PR TITLE
MTDSA-8977 Retrieve coding out RAML draft

### PR DESCRIPTION
**Pull request by RachGibbons**

### CODE REVIEW

File: resources/public/api/conf/1.0/retrieveCodingOutUnderpayments.raml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+is:
+  - headers.acceptJson
+  - headers.contentTypeJson
```
Comment (HarryBaxter) [inline resources/public/api/conf/1.0/retrieveCodingOutUnderpayments.raml:3]: Won't need contentTypeJson as its a GET

File: resources/public/api/conf/1.0/queryParameters.raml [hunk 2]
```diff
@@ -28,3 +28,8 @@ traits:
         description: 'To date to return charges.<br><br> Date in the format: YYYY-MM-DD'
         example: "2019-06-12"
         required: true
+  view:
+    queryParameters:
+      to:
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/queryParameters.raml:33]: this should be `source`

File: resources/public/api/conf/1.0/queryParameters.raml [hunk 3]
```diff
@@ -28,3 +28,8 @@ traits:
         description: 'To date to return charges.<br><br> Date in the format: YYYY-MM-DD'
         example: "2019-06-12"
         required: true
+  view:
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/queryParameters.raml:31]: also this

File: resources/public/api/conf/1.0/queryParameters.raml [hunk 4]
```diff
@@ -28,3 +28,8 @@ traits:
...
+    queryParameters:
+      to:
+        description: 'The type of business the loss applies to. Limited to the following possible values: <b><ul><li>`hmrcHeld`<li>`user`<li>`latest`</b>'
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/queryParameters.raml:34]: This description isn't correct

File: resources/public/api/conf/1.0/examples/retrieveCodingOutUnderpayments/retrieve_coding_out_underpayments_response.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+  "source": "hmrcHeld",
+  "taxCodeComponents": {
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/examples/retrieveCodingOutUnderpayments/retrieve_coding_out_underpayments_response.json:3]: this object doesn't appear on the response to vendors

File: resources/public/api/conf/1.0/examples/retrieveCodingOutUnderpayments/retrieve_coding_out_underpayments_response.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    "inYearAdjustments": {
+      "amount": 99.99,
+      "relatedTaxYear": "2021-22",
+      "submittedOn": "2021-07-06T09:37:17Z"
+    },
+    "links": [
+      {
+        "href": "/accounts/self-assessment/{nino}/{taxYear}/collection/tax-code",
+        "method": "PUT",
+        "rel": "create-and-amend-coding-out-underpayments"
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/examples/retrieveCodingOutUnderpayments/retrieve_coding_out_underpayments_response.json:34]:
```suggestion
        "rel": "create-or-amend-coding-out-underpayments"
```

File: resources/public/api/conf/1.0/errors.raml [hunk 7]
```diff
@@ -12,7 +12,7 @@ traits:
             type: types.errorResponse
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/errors.raml:12]: The error descriptions in this file are quite old. Can you update them to match the spec for retrieve coding out underpayments please?

File: resources/public/api/conf/1.0/schemas/retrieve_coding_out_underpayments_response.json [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+          "type": "string",
+          "example": "2021-22"
+        },
+        "submittedOn": {
+          "description": "The date the coding out information was added. It is either the latest change date held within HMRC or the latest user submitted date. It is in the format YYYY-MM-DDThh:mm:ssZ.",
+          "type": "string",
+          "example": "2020-07-06T09:37:17Z"
+        }
+      }
+    },
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/schemas/retrieve_coding_out_underpayments_response.json:38]:
```suggestion
"selfAssessmentUnderPayments": {
      "type": "array",
      "description": "Array containing details of coding out Self Assessment underpayments.",
      "items": {
        "type": "object",
        "properties": {
          "amount": {
            "description": "The total coding out amount attributed to Self Assessment underpayments. The value must be between 0 and 99999999999.99 up to 2 decimal places.",
            "type": "number",
            "multipleOf": 0.01,
            "minimum": 0.00,
            "maximum": 99999999999.99,
            "example": "5000.99"
          },
          "relatedTaxYear": {
            "description": "The tax year related to the underpayment.",
            "type": "string",
            "example": "2021-22"
          },
          "submittedOn": {
            "description": "The date the coding out information was added. It is either the latest change date held within HMRC or the latest user submitted date. It is in the format YYYY-MM-DDThh:mm:ssZ.",
            "type": "string",
            "example": "2020-07-06T09:37:17Z"
          }
        },
        "required": [
          "amount",
          "relatedTaxYear",
          "submittedOn"
        ]
      }
    },
```

For the array fields to show up properly in the RAML they need to be within a "properties" object like this. They should also be mandatory I think according to the spec. This applies to all of the arrays in this schema.

**Review by BetaDraconis — CHANGES_REQUESTED** [review]

File: resources/public/api/conf/1.0/schemas/retrieve_coding_out_underpayments_response.json [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        },
+        "submittedOn": {
+          "description": "The date the coding out information was added. It is either the latest change date held within HMRC or the latest user submitted date. It is in the format YYYY-MM-DDThh:mm:ssZ.",
+          "type": "string",
+          "example": "2020-07-06T09:37:17Z"
+            }
+          }
+        }
+      }
+    }
```
Comment (BetaDraconis) [inline resources/public/api/conf/1.0/schemas/retrieve_coding_out_underpayments_response.json:112]: HATEOAS links also need to be included here, like: https://github.com/hmrc/self-assessment-accounts-api/blob/a52324a272af32544fe7fa9da3b635bf5153c8f1/resources/public/api/conf/1.0/schemas/create_or_amend_coding_out_underpayments_response.json#L7

**Pull request merged**